File: blog-backend/src/main/java/com/example/blogbackend/services/category/CategoryService.java
package com.example.blogbackend.services.category;

import com.example.blogbackend.dtos.CategoryDTO;
import com.example.blogbackend.dtos.CategoryRequestDTO;

import java.util.List;
import java.util.UUID;

public interface CategoryService {
    List<CategoryDTO> getAllCategories();

    CategoryDTO getCategoryById(UUID id);

    CategoryDTO createCategory(CategoryRequestDTO categoryRequestDTO);

    void deleteCategory(UUID id);
}

File: blog-front/src/api/categoryService.js
import axiosInstance from "../utils/axiosInstance";

const API_URL = "/categories"; // Base URL for category endpoints

export const getAllCategories = async () => {
  try {
    const response = await axiosInstance.get(API_URL);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const getCategoryById = async (categoryId) => {
  try {
    const response = await axiosInstance.get(`${API_URL}/${categoryId}`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const createCategory = async (categoryRequestDTO) => {
  try {
    const response = await axiosInstance.post(API_URL, categoryRequestDTO);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const deleteCategory = async (categoryId) => {
  try {
    await axiosInstance.delete(`${API_URL}/${categoryId}`);
  } catch (error) {
    throw error;
  }
};
